utils/release_helper.py

```diff
@@ -0,0 +1,790 @@
+            for chars in ['{+', '+}']:
+                line = line.replace(chars, '')
+            new_contributors.append(line)
+    return '\n'.join(new_contributors)
+
+
+def is_contributors_list_updated(date_string) -> bool:
+    date = datetime.strptime(date_string, '%Y-%m-%d %H:%M')
+    # As a rule of thumbs, this function consider contributors list
+    # updated if not older than 2 weeks.
+    two_weeks_back = datetime.now() - timedelta(days=15)
+    if date < two_weeks_back:
+        return False
+    return True
+
+
+def update_contributors():
+    last_update = get_contributors_last_update()
+    # Currently, this script considers contributors lists is already updated if not
+    # older than 2 weeks.
+    if is_contributors_list_updated(last_update):
+        print(f'It is all fine, the contributors list was updated in {last_update}')
+    else:
+        print(f'Contributors list last update was in {last_update}. I can update it for you.')
+        utils_scripts_path = get_script_path()
+        contributors_script = os.path.join(utils_scripts_path, 'generate_contributors.py')
+        subprocess.run(
+            [f'PYTHONPATH=. {contributors_script}'], shell=True, cwd=get_repo_root_path())
+        show_git_diff()
+
+
+# Repository Milestones
+def has_milestone_version_format(milestone) -> bool:
+    regex = re.compile(r'\d\.\d\.\d')
+    return regex.match(milestone.title)
+
+
+def filter_version_milestones(milestones) -> list:
+    version_milestones = []
+    for milestone in milestones:
+        if has_milestone_version_format(milestone):
+            version_milestones.append(milestone)
+    return version_milestones
+
+
+def get_repo_milestones(repo) -> list:
+    return repo.get_milestones(state="all", direction="desc")
+
+
+def get_latest_version_milestone(repo) -> object:
+    milestones = get_repo_milestones(repo)
+    version_milestones = filter_version_milestones(milestones)
+    latest_version_milestone = version_milestones[0]
+    # It is not ensured the returned milestones list is ordered as expected.
+    # This function ensures the last milestone based on titles.
+    for milestone in version_milestones:
+        if milestone.title > latest_version_milestone.title:
+            latest_version_milestone = milestone
+    return latest_version_milestone
+
+
+def get_version_milestone(repo, version) -> object:
+    milestones = get_repo_milestones(repo)
+    version_milestones = filter_version_milestones(milestones)
+    for milestone in version_milestones:
+        if milestone.title == version:
+            return milestone
+
+
+def close_milestone(milestone) -> None:
+    if milestone.state == "closed":
+        print(f'Nice. The "{milestone}" milestone is already closed.')
+        return
+
+    if get_confirmation(f'The "{milestone}" milestone should be closed. Ok?'):
+        try:
+            milestone.edit(state="closed")
+        except Exception as e:
+            print(f'Error: {e}')
+            exit(1)
```

Comment:
I still don't like the fact that calling a function can lead to exit from the whole program. I think that it is fine to let an exception propagate upwards, and do the exit decision on the topmost level of the program - in main.

Reply:
Following the flow, after calling the `close_milestone` function,  the new milestone should be created by calling the `create_repo_milestone` function, but I don't want to proceed creating the new milestone if, by any reason, the old milestone couldn't be closed. The flow is intentionally and conservatively interrupted for investigation. I also note here that even if the milestone is already closed, this won't generate an exception. So, my point here is aligned to the the other comment in `update_milestone`. I don't think it is necessary to continue the flow and wait to report this on the topmost level. I also wouldn't like to do significant changes in the flow now.

Reply:
I agree with the behavior, but disagree with the implementation. If a function has a job to close a milestone, then it should either do it, or report to the caller if something went sideways. Imagine that you want to debug the script behavior in e.g. in an ipython session, where you call this function with an argument - you can get kicked out of the entire session.
"Somebody" from the chain of callers should decide how to react, and I agree that exiting like this is a good result. However, the decision to handle certain situation by exiting the whole program should be made on the `__main__` level, not below it. And isn't it true that exceptions provide a way of accomplishing it without the need to write any excessive code?
Anyway, as the script has a well-defined purpose and its code isn't shared, I don't insist that you change the behavior in order to be merged, but it is true that I consider the approach unfortunate.

Reply:
> I agree with the behavior, but disagree with the implementation. If a function has a job to close a milestone, then it should either do it, or report to the caller if something went sideways. 

The function already ensure this.

> Imagine that you want to debug the script behavior in e.g. in an ipython session, where you call this function with an argument - you can get kicked out of the entire session. "Somebody" from the chain of callers should decide how to react, and I agree that exiting like this is a good result. 

Currently, in the context of this script, there is not realistic case to handle differently any exception for this function. If there is a real demand in the future, it can be adapted.

> However, the decision to handle certain situation by exiting the whole program should be made on the `__main__` level, not below it. And isn't it true that exceptions provide a way of accomplishing it without the need to write any excessive code? 

This function is not called at the `__main__` level and your suggestion, as I mentioned, would require changes to the process flow. I'd rather change the process flow based on optimizations and then adapt the code for that optimization, rather than the other way around.

> Anyway, as the script has a well-defined purpose and its code isn't shared, I don't insist that you change the behavior in order to be merged, but it is true that I consider the approach unfortunate.

Thanks. It's good that we have different opinions on this. Neither yours nor mine are wrong, but I prefer to stick with my choice because I consider it better for this context.